### ISSUE TITLE
rabbitmq-common depends on xmerl

### ISSUE THREAD

**Issue by jlouis** [rabbitmq/rabbitmq-common#61]
Running a simple git-grep in rabbitmq-common shows it uses xmerl:

```
tiefling:rabbitmq-common jlouis$ gg xmerl
src/mochijson2.erl:197:            json_encode_string_unicode(xmerl_ucs:from_utf8(L), State, [?Q])
src/mochijson2.erl:204:            json_encode_string_unicode(xmerl_ucs:from_utf8(B), State, [?Q])
src/mochijson2.erl:297:                   [xmerl_ucs:to_utf8(C) | Acc];
src/mochijson2.erl:457:                [CodePoint] = xmerl_ucs:from_utf16be(<<C:16/big-unsigned-integer,
src/mochijson2.erl:459:                Acc1 = lists:reverse(xmerl_ucs:to_utf8(CodePoint), Acc),
src/mochijson2.erl:462:                Acc1 = lists:reverse(xmerl_ucs:to_utf8(C), Acc),
src/mochijson2.erl:713:        Expect = list_to_binary(xmerl_ucs:to_utf8(CodePoint)),
src/mochijson2.erl:748:    UTF8Seq = list_to_binary(xmerl_ucs:to_utf8(16#0001d120)),
src/mochijson2.erl:780:       iolist_to_binary(encode(list_to_atom(xmerl_ucs:to_utf8(16#0001d120))))),
src/mochijson2.erl:825:       json_bin_is_safe(list_to_binary(xmerl_ucs:to_utf8(16#0001d120)))),
src/mochijson2.erl:828:       xmerl_ucs:from_utf8(
src/mochijson2.erl:830:           decode(encode(list_to_atom(xmerl_ucs:to_utf8(16#0001d120))))))),
src/mochijson2.erl:836:       json_bin_is_safe(list_to_binary(xmerl_ucs:to_utf8([16#110000])))),
src/rabbit_binary_parser.erl:174:        xmerl_ucs:from_utf8(Bin),
```

But the .app file which is generated doesn't contain a reference to xmerl. The amqp_client now does and this avoids breaking releases, but this is a minor nitpick which could help getting fixed in order to make, e.g., dependency mappers more efficient.

**Comment by michaelklishin** [comment]
@jlouis in other words, you suggest that we make the `xmlerl` dependency explicit, correct?

**Comment by jlouis** [comment]
Yes, `xmerl` though (I always mistype it too :).

Just because the amqp_client library is using it and includes xmerl, you should probably include it here as well. Prior to 3.6.0, you get releases built without xmerl because none of `rabbitmq-common` or the `amqp-erlang-client` contains a reference to `xmerl`.

**Comment by michaelklishin** [comment]
Fixed by @binarin in #124. Thank you!
